python/cugraph/cugraph/community/louvain.py

```diff
@@ -89,6 +93,17 @@ def louvain(G, max_level=None, max_iter=None, resolution=1.0, threshold=1e-7):
 
     """
 
+    # FIXME: Onece the graph construction calls support isolated vertices through
+    #  the C API (the C++ interface already supports this) then there will be
+    # no need to compute isolated vertices here.
+
+    isolated_vertices = list()
+    if is_nx_graph_type(type(G)):
+        isolated_vertices = [v for v in range(G.number_of_nodes()) if G.degree[v] == 0]
```

Comment:
Like Brad probably mentioned. This may hurt our performance any time a user passes a networkX graph and this check will always be True whether we have isolated vertices or not. But it is ok for now since we don't have any alternative.